src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs

```diff
@@ -311,7 +311,10 @@ private void WriteFinalRecords()
             byte[] emptyRecord = new byte[TarHelpers.RecordSize];
             _archiveStream.Write(emptyRecord);
             _archiveStream.Write(emptyRecord);
-            _archiveStream.SetLength(_archiveStream.Position);
+            if (_archiveStream.CanSeek)
+            {
+                _archiveStream.SetLength(_archiveStream.Position);
```

Comment:
What is the purpose of the SetLength call?

Reply:
I think I can remove it completely.

I was trying to find the place where I initially added: I recall at the very beginning of writing this code, the `FileStream` was flushing the data to disk and writing it with more bytes than expected. Adding `SetLength` helped me fix that. But I don't have an argument right now to defend that behavior, the tests pass after removing the line.

Reply:
Cool.